docs/ai/quickstarts/build-mcp-server-publish-nuget.md

```diff
@@ -0,0 +1,220 @@
+
+1. Update the `<PackageId>` in the `.csproj` file to be unique on NuGet.org.
+
+2. Update your `.mcp/server.json` file with your respective MCP configuration details.
+
+    ```json
+    {
+      "description": "<your description here>",
+      "name": "io.github.<your GitHub username here>/<your repo name>",
+      "packages": [
+        {
+          "registry_name": "nuget",
+          "name": "<your package ID here>",
+          "version": "0.1.0-beta",
+          "package_arguments": [
+            {
+              "description": "Start the MCP server tool in server mode",
+              "is_required": true,
+              "format": "string",
+              "value": "start-mcp",
+              "default": "start-mcp",
+              "type": "positional",
+              "value_hint": "start-mcp"
+            }
+          ],
+          "environment_variables": [
+            {
+              "description": "The maximum number to return from the random number generator",
+              "name": "MAX_RANDOM_NUMBER"
+            }
+          ]
+        }
+      ],
+      "repository": {
+        "url": "https://github.com/<your GitHub username here>/<your repo name>",
+        "source": "github"
+      },
+      "version_detail": {
+        "version": "0.1.0-beta"
+      }
+    }
+    ```
+
+3. Pack the project:
+
+    ```bash
+    dotnet pack -c Release
+    ```
+
+4. Publish the package to NuGet:
+
+    ```bash
+    dotnet nuget push <path-to-nupkg-file> --source nuget.org --api-key <your-api-key>
+    ```
+
+See [Publish a package](https://learn.microsoft.com/nuget/nuget-org/publish-a-package) for more information.
+
+## Discover MCP servers on NuGet.org
+
+1. Browse for your MCP Server package on [NuGet.org](https://www.nuget.org/packages?q=&includeComputedFrameworks=true&packagetype=mcpserver&prerel=true&sortby=relevance).
+
+2. View the MCP Server details and copy the json.
+
+3. In your `mcp.json` file in the `.vscode` folder, add the following:
+
+    ```json
+    {   
+      "servers": {
+          "MinimalMcpServer": {
+              "type": "stdio",
+              "command": "dotnet",
+              "args": [ "tool",
+              "exec",
+              "<your-package-id>",
+              "--version",
+              "0.0.1-beta",
+              "--yes",
+              "--",
+              "start-mcp"
+
+              ],
+              "env": {
+                  "MAX_RANDOM_NUMBER": 100
+              }
+          }
+      }
+    }
+    ```
+
+4. Save the file.
+
+5. In GitHub Copilot, select the **Select tools** icon to verify your **MinimalMcpServer** is available with the tools listed.
+
+6. Enter a prompt to run the **GetRandomNumber** tool:
+
+    ```console
+    Give me a random number between 1 and 100.
+    ```
+
+7. Verify that the server responds with the echoed message:
+
+    ```output
+    Your random number is 9.
+    ```
+
+## Related content
+
+[Build a minimal MCP server](build-mcp-server.md)
```

Comment:
This should probably be a bulleted list, otherwise it's all put on one line (you can see that [here](https://learn.microsoft.com/en-us/dotnet/ai/quickstarts/build-mcp-server#related-content) where it should be fixed too @alexwolfmsft).

Reply:
added!